### DISCUSSION TITLE
Hosting App online

### DISCUSSION THREAD

**Discussion by mnallamalli97** [3Alan/DocsMind#17] [Q&A]
Hey @3Alan , Great work. You definitely deserve that coffee on me, I would love to contribute so you can keep your hard work. 

Can you please update the readme for step-by-step process on hosting this app online. Its quite confusing. If you could provide clearer instructions that would be awesome. Im trying to host this app for my high school to use and would like to give them a simple url for students to access and use.

PS: Im working on hooking this up with Google Drive folder so users can search their existing documents on Drive.

**Comment by 3Alan** [comment]
Thank you very much for your recognition. As I don't have a credit card, I can only deploy my demo on cloud service providers that offer free trial. Vercel is used to deploy the UI and Railway is used to deploy the backend service (which does not support file persistence). This is also why user uploads are not supported in the demo. If you want to apply it to a production environment, you will need a server or any cloud service provider that supports Docker compose to run this application. How to use Docker to run the application has been explained in README.

**Reply by beralc** [reply]
Hi @3Alan , You'll definitely be getting a coffee from me too!! 
I managed to install docker on aws, and follow the instructions. However, the backend starts in 127.0.0.1:500 and vite in http://localhost:5173 even though i put the ip address of the server where I would like to run the app in the .env file. Here is a screenshot: 
<img width="1130" alt="image" src="https://github.com/3Alan/DocsMind/assets/10000606/dfed47e5-90c7-48fa-aecc-6725407b4ec1">

What am i doing wrong?

**Comment by mnallamalli97** [comment]
@3Alan 
Update for you. I was able to host the app on Vercel (/client) and on Railway (/server)
Now, when I try to upload a file I get: 

<img width="752" alt="Screenshot 2023-05-09 at 4 56 14 PM" src="https://github.com/3Alan/DocsMind/assets/21005221/00dc0a53-fea6-4398-904a-4e153ac2c11d">


Is there anything I need to do to the .env file or anything? 
If I need to update it, what do I need to add? How do I find the VITE_SERVICES_URL on Railway?

**Reply by mnallamalli97** [reply]
@3Alan 
Its built on Railway now, and I updated the .env file to look like this: 

<img width="530" alt="Screenshot 2023-05-09 at 6 53 20 PM" src="https://github.com/3Alan/DocsMind/assets/21005221/12bb39d9-9ada-47e1-9330-6fddb808a856">


but this is the error when I upload a file: 

<img width="833" alt="Screenshot 2023-05-09 at 6 12 51 PM" src="https://github.com/3Alan/DocsMind/assets/21005221/53edfa97-9630-4543-a550-15ad607285bd">


Not sure why its still taking `http://localhost:8080/api/upload`

Am I doing something wrong here?

**Reply by 3Alan** [reply]
If your UI is deployed on Vercel, then you need to configure environment variables in Vercel.
![image](https://github.com/3Alan/DocsMind/assets/49978973/b3fe055b-e500-4f8b-adf2-f49280cf03ac)

**Comment by 3Alan** [comment]
I currently don't have any good recommendations for free trial cloud service providers. Railway cannot persist data because the project does not include a database and all data is saved locally. Therefore, once you redeploy, the files you previously uploaded will be destroyed. Currently, the only option is to use Docker to deploy the application to your private VPS.

**Comment by mnallamalli97** [comment]
So I have my Railway running. 

I added the VITE SERVICES URL to the Vercel variables:
<img width="1011" alt="Screenshot 2023-05-10 at 9 49 35 PM" src="https://github.com/3Alan/DocsMind/assets/21005221/aa9dd3e6-a1af-4f52-9d03-6353371c01d8">

Here is the Railway server that is curently being listened to: 
<img width="632" alt="Screenshot 2023-05-10 at 9 50 11 PM" src="https://github.com/3Alan/DocsMind/assets/21005221/f18687a4-0c41-409e-82e0-b5079a411212">

This is the error when I upload a file:
<img width="849" alt="Screenshot 2023-05-10 at 9 48 38 PM" src="https://github.com/3Alan/DocsMind/assets/21005221/ddbdccf8-3c1f-4aec-97a1-68bd160b29c7">

**Comment by 3Alan** [comment]
I have updated the [documentation](https://github.com/3Alan/DocsMind/blob/main/Deployment.md) on deployment.
